### PR TITLE
Feature/#67

### LINKED ISSUE #67: Implement User Service
- create
- delete
- update

### PR DESCRIPTION
**Pull request by Howrran**

### CODE REVIEW

File: src/app/services/user.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        user = User(username=username, email=email, google_token=google_token)
+        DB.session.add(user)
+        return user
+
+    @staticmethod
+    @transaction_decorator
+    def get_by_id(user_id):
+        """
+        Get user by id
+        :param id:
```
Comment (yurdosii) [inline src/app/services/user.py]: miss newline between "Get user by id" and ":param"

File: src/app/services/user.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        :param email:
+        :param google_token:
+        :return: user or None
+        """
+        user = User.query.filter_by(email=email,
+                                    google_token=google_token).first()
+        if not user:
+            return None
+
+        return user
```
Comment (IvanKostiv) [inline src/app/services/user.py]: Refactor filter

File: src/app/services/user.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        :param user_id:
+        :return: True or None
+        """
+        user = UserService.get_by_id(user_id)
+
+        if user:
+            DB.session.delete(user)
+            return True
+
+        return None
```
Comment (IvanKostiv) [inline src/app/services/user.py]: Create exception UserNotExist and raise if UserService.get_by_id is None

File: src/app/services/user.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+        :param username:
+        :param email:
+        :param google_token:
+        :return: user or none
+        """
+        user = UserService.get_by_id(user_id)
+
+        if not user:
+            return None
```
Comment (IvanKostiv) [inline src/app/services/user.py]: Raise Error UserNotExist

File: src/app/services/user.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+    @staticmethod
+    @transaction_decorator
+    def update(user_id, username=None, email=None, google_token=None):
+        """
+        Update user info in database
+
+        :param username:
+        :param email:
+        :param google_token:
```
Comment (yurdosii) [inline src/app/services/user.py]: Where is user_id as param?

File: src/app/services/user.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+        if user is None:
+            raise UserNotExist()
+
+        DB.session.delete(user)
+        return True
+
+    @staticmethod
+    @transaction_decorator
+    def user_filter(username=None, email=None, google_token=None):
```
Comment (IvanKostiv) [inline src/app/services/user.py]: The method name should be just a "filter"

File: src/app/services/user.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        data = {}
+
+        if username:
+            data['username'] = username
+
+        if email:
+            data['email'] = email
+
+        if google_token:
+            data['google_token'] = google_token
```
Comment (IvanKostiv) [inline src/app/services/user.py]: if ... is not None

File: src/app/services/user.py [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+        user = User.query.filter_by(**data).all()
+
+        if not user:
+            raise UserNotExist()
```
Comment (IvanKostiv) [inline src/app/services/user.py]: You just return empty list, not raise error

File: src/app/services/user.py (hunk 7 above)
Comment (yurdosii) [reply src/app/services/user.py]: extra new lines between ifs

File: src/app/services/user.py (hunk 8 above)
Comment (Howrran) [reply src/app/services/user.py]: fixed and replied :)

**Review by AndriiKret — DISMISSED** [review]

**Review by zzeropilot — DISMISSED** [review]

**Review by AndriiKret — APPROVED** [review]

**Review by zzeropilot — APPROVED** [review]

File: src/app/services/__init__.py [hunk 9]
```diff
@@ -3,3 +3,5 @@
 """
 
 from .field_range import FieldRangeService
+
+from .user import UserService
```
Comment (yurdosii) [inline src/app/services/__init__.py:7]: extra new line

File: src/app/services/user.py [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        :param email:
+        :param google_token:
+        :return: user or None
+        """
+
+        data = {}
+
+        if username is not None:
+            data['username'] = username
+
```
Comment (yurdosii) [inline src/app/services/user.py:110]: extra new line

File: src/app/services/user.py [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+        if email is not None:
+            data['email'] = email
+
```
Comment (yurdosii) [inline src/app/services/user.py:113]: extra new line

**Review by dtymkiv — APPROVED** [review]

File: src/app/services/user.py [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+        if google_token is not None:
+            data['google_token'] = google_token
+
+        user = User.query.filter_by(**data).all()
```
Comment (yurdosii) [inline src/app/services/user.py:117]: Rename 'user' to 'users'

**Comment by yurdosii** [comment]
Check comments

**Pull request merged**